bindings/python/benchmark.py

```diff
@@ -0,0 +1,202 @@
+raw_bson = None
+large_doc_keys = None
+collection_names = {LARGE: "large", SMALL: "small"}
+dtypes = {}
+schemas = {}
+raw_bsons = {}
+
+
+def _setup():
+    global db
+    global raw_bson
+    global large_doc_keys
+
+    db = pymongo.MongoClient().pymongoarrow_test
+    small = db[collection_names[SMALL]]
+    small.drop()
+
+    print("%d small docs, %d bytes each with 3 keys" % (
+        N_SMALL_DOCS,
+        len(BSON.encode({'_id': ObjectId(), 'x': 1, 'y': math.pi}))))
+
+    small.insert_many([
+        collections.OrderedDict([('x', 1), ('y', math.pi)])
+        for _ in range(N_SMALL_DOCS)])
+
+    dtypes[SMALL] = np.dtype([('x', np.int64), ('y', np.float64)])
+    schemas[SMALL] = Schema({'x': pyarrow.int64(), 'y': pyarrow.float64()})
+
+    large = db[collection_names[LARGE]]
+    large.drop()
+    # 2600 keys: 'a', 'aa', 'aaa', .., 'zz..z'
+    large_doc_keys = [c * i for c in string.ascii_lowercase
+                      for i in range(1, 101)]
+    large_doc = collections.OrderedDict([(k, math.pi) for k in large_doc_keys])
+    print("%d large docs, %dk each with %d keys" % (
+        N_LARGE_DOCS, len(BSON.encode(large_doc)) // 1024, len(large_doc_keys)))
+
+    large.insert_many([large_doc.copy() for _ in range(N_LARGE_DOCS)])
+
+    dtypes[LARGE] = np.dtype([(k, np.float64) for k in large_doc_keys])
+    schemas[LARGE] = Schema({k: pyarrow.float64() for k in large_doc_keys})
+
+    # Ignore for now that the first batch defaults to 101 documents.
+    raw_bson_docs_small = [{'x': 1, 'y': math.pi} for _ in range(N_SMALL_DOCS)]
+    raw_bson_small = BSON.encode({'ok': 1,
+                                  'cursor': {
+                                      'id': Int64(1234),
+                                      'ns': 'db.collection',
+                                      'firstBatch': raw_bson_docs_small}})
+
+    raw_bson_docs_large = [large_doc.copy() for _ in range(N_LARGE_DOCS)]
+    raw_bson_large = BSON.encode({'ok': 1,
+                                  'cursor': {
+                                      'id': Int64(1234),
+                                      'ns': 'db.collection',
+                                      'firstBatch': raw_bson_docs_large}})
+
+    raw_bsons[SMALL] = raw_bson_small
+    raw_bsons[LARGE] = raw_bson_large
+
+
+def _teardown():
+    db.collection.drop()
+
+
+bench_fns = collections.OrderedDict()
+
+
+def bench(name):
+    def assign_name(fn):
+        assert fn not in bench_fns.values()
+        bench_fns[name] = fn
+        return fn
+
+    return assign_name
+
+
+@bench('conventional-to-ndarray')
+def conventional_ndarray(use_large):
+    collection = db[collection_names[use_large]]
+    cursor = collection.find()
+    dtype = dtypes[use_large]
+
+    if use_large:
+        np.array([tuple(doc[k] for k in large_doc_keys) for doc in cursor],
+                 dtype=dtype)
+    else:
+        np.array([(doc['x'], doc['y']) for doc in cursor], dtype=dtype)
+
+
+@bench('pymongoarrow-to-numpy')
```

Comment:
I named it `to-numpy` because the output isn't an ndarray. Should I change the `conventional_ndarray` method to output the same format as find_numpy_all?

Reply:
Adding a comment stating clearly what the discrepancy is will be enough.

Reply:
Done.